core/src/main/scala/cats/Traverse.scala

```diff
@@ -168,6 +168,24 @@ trait Traverse[F[_]] extends Functor[F] with Foldable[F] with UnorderedTraverse[
   def zipWithIndex[A](fa: F[A]): F[(A, Int)] =
     mapWithIndex(fa)((a, i) => (a, i))
 
+  /**
+    * Same as [[traverseWithIndexM]] but the index type is [[Long]] instead of [[Int]].
+    */
+  def traverseWithLongIndexM[G[_], A, B](fa: F[A])(f: (A, Long) => G[B])(implicit G: Monad[G]): G[F[B]] =
+    traverse(fa)(a => StateT((s: Long) => G.map(f(a, s))(b => (s + 1, b)))).runA(0L)
+
+  /**
+    * Same as [[mapWithIndex]] but the index type is [[Long]] instead of [[Int]].
+    */
+  def mapWithLongIndex[A, B](fa: F[A])(f: (A, Long) => B): F[B] =
+    traverseWithLongIndexM[cats.Id, A, B](fa)((a, long) => f(a, long))
+
+  /**
+    * Same as [[zipWithIndex]] but the index type is [[Long]] instead of [[Int]].
+    */
+  def zipWithLongIndex[A](fa: F[A]): F[(A, Long)] =
+    traverseWithLongIndexM[cats.Id, A, (A, Long)](fa)((a, long) => (a, long))
```

Comment:
I wonder if this can be implemented via `mapWithLongIndex`, so that the `zipWithLongIndex` gets the performance gains in collections that have overridden `mapWithLongIndex` (e.g.  https://github.com/typelevel/cats/pull/4247/commits/6fe035f533b7ff1d46eb297c4b2d6b9954dcdd53)

Reply:
turns out I can lol (https://github.com/typelevel/cats/pull/4247/commits/609a4f1958db09ca1781042d95a09dbc78d9e08b)